### DISCUSSION TITLE
Raspbian low startup

### DISCUSSION THREAD

**Discussion by Giacomo-Gozzi** [AvaloniaUI/Avalonia#7177] [General]
I created a new application from Visual studio, I use NET Core 6, starting the application on Raspberry takes about 30 seconds, how can I do to reduce this time? Once started the application works without any problem even the CPU usage is very low.

**Comment by ahopper** [comment]
Publishing with /p:PublishReadyToRun=true can help.  30s does seem very slow, what rpi is that on?

**Reply by Giacomo-Gozzi** [reply]
Even with that setting I tried the time does not change.
You can easily try it too by creating a new empty project from visual studio, with the latest version of raspbian and a raspberry pi3 or pi4

**Comment by ahopper** [comment]
I Just started up my non trivial app on rpi4 and it took about 10s to appear, this was built with Avalonia 0.10.7 and .net5.0, I'll try a .net 6.0 and latest Avalonia build see if that makes it worse.

**Reply by Giacomo-Gozzi** [reply]
Could you send me the command line you use to compile the app?
And the configuration of avalonia?
So I can try with mine if I can solve the problem.

I attach my configuration

![image](https://user-images.githubusercontent.com/22610788/146564006-de8f8f27-ec5b-43ce-8d30-b09d94fd795a.jpeg)

**Comment by ahopper** [comment]
```dotnet publish -r linux-arm -c Release -f net6.0 /p:PublishTrimmed=true -p:TrimMode=Link /p:PublishReadyToRun=true /p:PubLin=True /p:PublishSingleFile=true```

I avoid the platform detect as in the past it made the app smaller, no idea if it is still true

```
#elif PUBLIN

            var builder = AppBuilder.Configure<App>()
                .UseX11()
                .UseReactiveUI()
                .UseManagedSystemDialogs()
                .With(new X11PlatformOptions { UseGpu = UseGpu, UseEGL = UseEgl, OverlayPopups = true  })
                .UseSkia();
```

**Reply by Giacomo-Gozzi** [reply]
The cpu utilization remains really high now and the time has improved by only 10 seconds

**Comment by ahopper** [comment]
What Avalonia version are you using?  I just tried 0.10.11-rc.2 and it does not run for me on arm32, I've probably done something silly and will try again later.

**Reply by ahopper** [reply]
I fixed my issues and it starts up a bit faster with .net6.0.1 and 0.10.11-rc.2   <8s . Cpu is close to zero with nothing happening and about what I expect when stuff is happening.  So what is different between our projects? I do have the opengl desktop enabled on the pi.

**Reply by Giacomo-Gozzi** [reply]
How can i use this version? On nuget i can find only 0.10.10?

**Reply by ahopper** [reply]
In vs click 'include prerelease' or just type the version in .csproj
